src/main/java/com/example/onlinebookstore/dto/book/CreateBookDto.java

```diff
@@ -1,13 +1,21 @@
 package com.example.onlinebookstore.dto.book;
 
+import com.example.onlinebookstore.model.Category;
+import jakarta.validation.constraints.Min;
 import java.math.BigDecimal;
+import java.util.Set;
 import lombok.Data;
+import lombok.NonNull;
 
 @Data
 public class CreateBookDto {
+    @NonNull
     private String title;
+    @NonNull
     private String author;
+    @Min(0)
     private BigDecimal price;
     private String description;
     private String coverImage;
+    private Set<Category> categoryIds;
```

Comment:
```suggestion
    private Set<Long> categoryIds;    
```
You have to get only the ids of the categories (or CategoryDto) Also name of field have to represent stored value

Reply:
it's dto for create book. It there should be the set of categories. Field is already renamed